azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -905,19 +911,15 @@ public void resume(String user) {
   }
 
   public void kill(String user) {
-    synchronized (mainSyncObj) {
-      logger.info("Flow killed by " + user);
-      flow.setStatus(Status.KILLED);
-      kill();
-      updateFlow();
-    }
-    interrupt();
+    logger.info("Flow killed by " + user);
+    kill();
   }
 
-  private void kill() {
+  public void kill() {
     synchronized (mainSyncObj) {
```

Comment:
checked the code a bit. runner.kill() calls job.cancel() which is implemented by jobtypes and does best effort killing, even if the cancel is not successful, runner.kill() marks the job as killed anyway. Similarly, the flow is marked as killed as long as user issues the kill command on it, but it doesn't guarantee the flow is actually killed.